src/main/scala/io/viash/config/package.scala

```diff
@@ -150,8 +159,88 @@ package object config {
+            case (None, _) => jo
+            case (Some(args), None) => 
+              jo.add("argument_groups", Json.fromValues(List(
+                Json.fromJsonObject(
+                  JsonObject(
+                    "name" -> Json.fromString("Arguments"),
+                    "arguments" -> args
+                  )
+                )
+              )))
+            case (Some(args), Some(arg_groups)) =>
+              // determine if we should prepend or append arguments to argument_groups
+              val prepend = jo.keys.find(s => s == "argument_groups" || s == "arguments") == Some("arguments")
+              def combinerSeq(a: Vector[Json], b: Seq[Json]) = if (prepend) b ++: a else a :++ b
+              def combiner(a: Vector[Json], b: Json) = if (prepend) b +: a else a :+ b
+
+              // get the argument group named 'Arguments' from arg_groups
+              val argumentsGroup = arg_groups.asArray.flatMap(_.find(_.asObject.exists(_.apply("name").exists(_ == Json.fromString("Arguments")))))
+              argumentsGroup match {
+                case None =>
+                  // no argument_group with name 'Argument' exists, so just add arguments as a new argument group
+                  jo.add("argument_groups", Json.fromValues(
+                    combiner(
+                      arg_groups.asArray.get,
+                      Json.fromJsonObject(
+                        JsonObject(
+                          "name" -> Json.fromString("Arguments"),
+                          "arguments" -> args
+                        )
+                      )
+                    )
+                  ))
+                case Some(ag) =>
+                  // argument_group with name 'Argument' exists, so add arguments to this argument group
+                  val newAg = ag.asObject.get.add("arguments",
+                    Json.fromValues(
+                      combinerSeq(
+                        ag.asObject.get.apply("arguments").get.asArray.get,
+                        args.asArray.get
+                      )
+                    )
+                  )
+                  jo.add("argument_groups", Json.fromValues(arg_groups.asArray.get.map{
+                    case ag if ag == argumentsGroup.get => Json.fromJsonObject(newAg)
+                    case ag => ag
+                  }))
+              }
+          }
+          Json.fromJsonObject(newJsonObject.remove("arguments"))
+      }    
+  }}
+  .prepare {
+    // Move functionality to config level, if functionality exists also move .info to .build_info
+    _.withFocus{
+      _.mapObject{ conf =>
+        conf.contains("functionality") match {
+          case true => 
+            val functionality = conf.apply("functionality").get.asObject.get
+            val buildInfo = conf.apply("info")
+            val conf1 = buildInfo.map{ bi => 
+                conf.remove("info").add("build_info", bi)
+              }.getOrElse(conf)
```

Comment:
Why move `info` to `build_info`? The config shouldn't contain a `info` field at this stage yet, right?

...Or are we using this function to read in `.config.vsh.yaml` files as well? :eyes:

Reply:
It is indeed meant to have *some* sort of compatibility with (old) `.config.vsh.yaml` files. I wanted to prevent poluting & mixing `.functionality.info` with the build info under `.info`. Unlikely to happen often, but it was easy to write too.
Also, the code should be scheduled to be dropped in the future, so any extra complexity here shoud be temporary.